File: reto_clean_code/angular/step1-procedural.ts
// Script procedural que utiliza variables globales para realizar una
// transferencia entre dos cuentas.  Este archivo sirve como punto de partida
// para refactorizar a un enfoque orientado a objetos.

let saldoCuentaA = 1000;
let saldoCuentaB = 500;

export function transferir(origen: 'A' | 'B', monto: number): void {
  if (origen === 'A') {
    if (saldoCuentaA >= monto) {
      saldoCuentaA -= monto;
      saldoCuentaB += monto;
      console.log('Transferencia exitosa.');
    } else {
      console.log('Saldo insuficiente.');
    }
  } else {
    if (saldoCuentaB >= monto) {
      saldoCuentaB -= monto;
      saldoCuentaA += monto;
      console.log('Transferencia exitosa.');
    } else {
      console.log('Saldo insuficiente.');
    }
  }
}

File: reto_clean_code/python/step1_procedural.py
"""
Script procedural para transferir fondos entre dos cuentas.  Define variables
globales y una función que modifica estas variables directamente.  Se utiliza
como base para refactorizar a un diseño orientado a objetos en el paso 1.
"""

saldo_cuenta_a = 1000
saldo_cuenta_b = 500


def transferir(origen: str, destino: str, monto: float) -> None:
    global saldo_cuenta_a, saldo_cuenta_b
    if origen == 'A':
        if saldo_cuenta_a >= monto:
            saldo_cuenta_a -= monto
            saldo_cuenta_b += monto
            print("Transferencia exitosa.")
        else:
            print("Saldo insuficiente.")
    elif origen == 'B':
        if saldo_cuenta_b >= monto:
            saldo_cuenta_b -= monto
            saldo_cuenta_a += monto
            print("Transferencia exitosa.")
        else:
            print("Saldo insuficiente.")


if __name__ == "__main__":
    transferir('A', 'B', 200)
    print(saldo_cuenta_a, saldo_cuenta_b)